### ISSUE TITLE
[Feature]“GetImplicitUsersForResource” cannot  support multiple “g”?

### ISSUE THREAD

**Issue by nengc** [casbin/casbin#1317]
**Want to prioritize this issue? Try:**

[![issuehunt-to-marktext](https://github.com/BoostIO/issuehunt-materials/raw/master/v1/issuehunt-button-v1.svg)](https://issuehunt.io/r/casbin/casbin)

------

**Is your feature request related to a problem? Please describe.**
I need to use "GetImplicitUsersForResource", but found only one "g" was supported. 
And the model i used like https://editor.casbin.org/#54C9CCD45

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by hsluoyz** [comment]
@PokIsemaine

**Comment by PokIsemaine** [comment]
@nengc Yes, this is the current drawback, and we're thinking about how to enhance the API in question. 
There are two main aspects: 
1. support for multiple `g` types (This will be achieved as soon as possible)
2. support for inheritance relationships such as `resources/obj`, `domains`, and so on (This may take a while)

**Comment by hsluoyz** [comment]
Need to add `GetNamedImplicitUsersForResource`() API

**Comment by D0000M** [comment]
#1523 has this feat, like
```
e, _ := casbin.NewEnforcer("examples/rbac_with_resource_roles_model.conf", "examples/rbac_with_resource_roles_policy.csv")
data2Users, err := e.GetNamedImplicitUsersForResource("g2", "data2")
```

**Comment by hsluoyz** [comment]
Closed as resolved
